testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -387,6 +434,21 @@ impl ClusterSwarmKube {
         Ok(workspace.clone())
     }
 
+    pub async fn spawn_job(
```

Comment:
Can you make `Instance::spawn_job` that will call into this function, and use it from client compat test?
In this case, you don't need to make `K8sInstanceInfo `, `K8sInstanceInfo::k8s_node` and `Instance::k8s_backend` public.